### PR TITLE
Bookstore: initialize project with components

### PR DESCRIPTION
**Pull request by leonardoeng13**
### In this project milestone I accomplished the following :
- [x] Set-up react app using react documentation.
- [x] Add React Redux
- [x] Structure your application files using a "feature folder"
- [x] Add React Router and set two s and s for the app's navigation (Books and Categories)
   - Books - the default view
      - Displayed the list of books (empty at this point but it should be ready for the data) with the Remove button (no functionality yet)
      - Had a form for adding a book (no functionality yet)
  - Categories
      -  Displayed "Under construction" text only.
- [x] Follow Gitflow requirements.
- [x] Set up my local environment to fix all linters errors.
- [x] Deploy the app with Netlify

_**Thank you in advance for your feedback**_

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
+{
+  "name": "bookstore",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
```
Comment (GzimAsani) [inline package.json:5]: `react-router-dom` should be present in package.json
`react-router-dom` will be needed to make the routes

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import './index.css';
+import App from './App';
+// import reportWebVitals from './reportWebVitals';
```
Comment (GzimAsani) [inline src/index.js]: - [ ] Remove the commented code

File: README.md [hunk 3]
```diff
@@ -1,67 +1,146 @@
...
+</p>
+
+</br>
+
+<!-- <div align="center">
+<kbd>
+  <img alt="math" src="https://user-images.githubusercontent.com/80895497/134748890-29d5e63a-e1e4-4438-b23b-244824012a17.png">
+  </kbd>
+  </div>
+ </br> -->
```
Comment (GzimAsani) [inline README.md]: - [ ] Remove commented code

File: README.md [hunk 4]
```diff
@@ -1,67 +1,146 @@
...
-![screenshot](./app_screenshot.png)
+## Built With
 
-Additional description about the project and its features.
+- JSX, JavaScript with ES6
+- React.js with redux
+- Linters, Node.js, Git, Flexbox, Bootstrap, Webpack
 
-## Built With
+# Getting Started with Create React App
```
Comment (GzimAsani) [inline README.md:50]: - [ ] Be more specific here. Explain how to clone the project

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# Hi @leonardoeng13   
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight:
- Good use of gitflow :+1: 

## Required Changes ♻️
- [ ] _Check the comments under the review for more information._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 4 above)
Comment (leonardoeng13) [reply README.md:50]: Do you know this part of the project it's just the components and don't have anything functional? Do you know this?

Comment (leonardoeng13) [reply README.md:50]: And the information is already there. Do you how to check the right branch?

File: package.json (hunk 1 above)
Comment (bellom) [reply package.json:5]: According to the project requirements, you're supposed to add `react-router-dom`
Screenshot from the requirement page:
image

**Comment by bellom** [comment]
Hello 👋,

This is an additional review as a result of the second opinion request.

@leonardoeng13 Keep in mind that there is a review guideline the reviewer followed which is also aligned with the project requirements, according to the [project requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/project_init_project_wth_comps.md#project-requirements) (bullet 4) you're supposed to add `react-router-dom`.
And also, as you can see above, the reviewer drop some inline comments on `feature/component` branch, I don't understand the question you asked the reviewer, from what I can see the reviewer checked the right branch, and you're missing the setup setting in your README file.

Example of a good setup setting:
```
- git clone "your repo"
- cd into the folder
- git checkout "branch-name"
- run "npm i"
- run "npm start"
```
@leonardoeng13 Please let's be more professional in our comments next time, code reviewers are doing their best to help and to correct students politely. 👍 

@GzimAsani, In your next review, please it would be nice to show the students they are wrong than telling them they are wrong, explain the reason why you're asking for changes, and also, you can always write an inline comment with screenshots, check my inline comment above.

@leonardoeng13 Kindly make all necessary changes and ask for a re-review.

Keep rocking 🚀!

File: README.md [hunk 5]
```diff
@@ -1,67 +1,153 @@
...
 
-- Major languages
-- Frameworks
-- Technologies used
+This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
 
-## Live Demo
+## How to run this proiject:
+- git clone "your repo"
```
Comment (GzimAsani) [inline README.md]: - [x] Add your repo inside parentheses**("")** in your case it should be "https://github.com/leonardoeng13/Bookstore.git". Keep in mind that your Readme may be red by people that are not that familiar with programming and you should try to make the steps in your readme file as simple as possible

File: README.md [hunk 6]
```diff
@@ -1,67 +1,153 @@
...
+- cd into the folder
+- git checkout "branch-name"
```
Comment (GzimAsani) [inline README.md:57]: - [x] Enter your branch name here

File: README.md (hunk 3 above)
Comment (GzimAsani) [reply README.md]: - [x] Please remove the commented code, it is not a good practice

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# Hi @leonardoeng13   
Good job so far!
Thank you for implementing the requirements but unfortunately, there are still some issues that you still need to work on to go to the next project but you are almost there!

## To highlight:
- Good use of gitflow :+1: 

## Required Changes ♻️
- [ ] _Check the comments under the review for more information._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         

## Thank you for implementing all the project requirements, you've done a great job :trophy:  . There is nothing else to say other than... it's time to merge it :champagne:  :tada: Congratulations

Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**